### DISCUSSION TITLE
copied icons from font web browser shows as ?

### DISCUSSION THREAD

**Discussion by SamiAlsubhi** [ryanoasis/nerd-fonts#815] [Q&A]
I have hack nerd font installed. I use it mainly for neovim. icons display correctly. However when i copy icons from https://www.nerdfonts.com/cheat-sheet and I paste it somewhere (mainly neovim) it shows the icon as '?'.
What is the issue here?
Icons that are already inside files I downloaded from the internet display correctly.
However when I copy paste them, they show as '?'

**Comment by Finii** [comment]
Hmm, unfortunately there is no information about your setup.

For me it works, here for example I copy the github mascot (click on `copy -> icon`) and pasted into `tilix`:

![image](https://user-images.githubusercontent.com/16012374/161266908-7f8b16a8-31fc-474d-96cf-73caeeaf3fac.png)

The only thing that I know of where `?` are shown instead of the symbols is in some Windows consoles like `Terminal` or `cmd`?
I believe it is a feature to alert you the stuff you inserted is not what you might think. What does work is `echo ?` (where `?` is the inserted symbol and not a question mark) and then after pressing enter you are allowed to see the actual icon :grimacing: 

Maybe you can give a concrete example with all the applications involved (try to simplify for me, an application that is on board by default is easiest for me to reproduce), with screenshot etc. Thank you.

**Reply by Finii** [reply]
Hmm, started up Windows, and it does not work (as you described) with `gVim` and `Windows Terminal`, but it did work in `git bash`. Hmm.

![image](https://user-images.githubusercontent.com/16012374/161269940-054cf888-b14b-46ba-b393-b8ce5b4a0fc3.png)

Are you on Windows?

**Reply by Finii** [reply]
Oh, it **does** work in `Windows Terminal`, I just forgot to push 'save' after changing the font :grimacing: 

![image](https://user-images.githubusercontent.com/16012374/161271934-b3ee5ed2-87c2-41ec-9635-85c382dac581.png)

**Reply by Finii** [reply]
Ah...

In `gVim` at least, you need to set the file encoding, or the multi byte paste will not work.

Open a file, enter `:set encoding=utf-8` `<ENTER>` and insert the stuff. (Keep in mind you need to change the encoding before you paste.)

![image](https://user-images.githubusercontent.com/16012374/161275400-0285a799-32bb-4508-95ec-c22ee3a69ee7.png)

**Reply by SamiAlsubhi** [reply]
Pasting into iterm2 terminal works fine.
Pasting into neovim using `p` keyboard key results in pasting '?'. (That's the default key to paste in neovim)
However, I just found out that if I am in `INSERT` mode and pressed `CTL-V`, it works just fine.

I tried setting `fileencoding` and `encoding` to UTF-8 but it does not solve the `p` key pasting issue. This is very weird.

**Reply by SamiAlsubhi** [reply]
I think as`CTL-V` works fine it should be OK as I rarely copy icons anyways.

**Reply by Finii** [reply]
When I'm in `neovim` (which is my everyday editor, as commandline application of course, not as a windowed appliaction), when I do `p` I paste the vi-internal buffer (i.e. stuff copied with `y` for example. It does not paste the systemwide buffer.

Furthermore, if I do a `i` `ctrl-v` I'm in protected input mode. That has always been the case back to `elvis` and the real `vi`, as far as I remember. So I can not paste in insert mode with `ctrl-v`.

This means I do not know what you do there. That is not standard (neo)vim.

What I found was this: https://ploegert.gitbook.io/til/tools/vim/allow-neovim-to-copy-paste-with-system-clipboard

So what I did was this:
* fire up `gVim`
* `:set encoding=utf-8`
* `:set clipboard=unnamed`
* copy some glyph from the cheat sheet in firefox
* `p`

works.

See
* https://neovim.io/doc/user/options.html#'clipboard'
* https://neovim.io/doc/user/provider.html#clipboard

_Edit: Add neovim docu links_

**Reply by SamiAlsubhi** [reply]
My Neovim is configured to have `encoding=utf-8` and `clipbaord={"unnamed","unnamedplus"}.`
https://github.com/SamiAlsubhi/nvim/blob/master/lua/user/options.lua
Nevertheless, I did your steps but pasted the icon as `?`.
I am on Mac, so my `ctrl` is `command`. When in insert mode ,`command-v` works and pastes the icon correctly.
It is a workaround of course and it is ok as I only copy icons to customise my neovim looks and that rarely happens. Therefore the workaround does the job.
